src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -68,25 +72,45 @@ public void addAnswer(Answer answer) {
         answers.add(answer);
     }
 
-    public boolean isOwner(NsUser loginUser) {
+    private boolean isOwner(NsUser loginUser) {
         return writer.equals(loginUser);
     }
 
-    public Question setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
+    public void delete(NsUser loginUser) {
+        if (!isOwner(loginUser)) {
+            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+        }
+        if (hasOtherUserAnswers()) {
+            throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
+        }
+        this.deleted = true;
+        this.deletedDate = LocalDateTime.now();
```

Comment:
```suggestion
    public void delete(NsUser loginUser, LocalDateTime deletedDate) {
        if (!isOwner(loginUser)) {
            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
        }
        if (hasOtherUserAnswers()) {
            throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
        }
        this.deleted = true;
        this.deletedDate = deletedDate;
```

`delete()` 메소드를 테스트하기 쉽게 만들기위해
`deletedDate`를 외부에서 주입받는 방식으로 시그니처를 변경해보는 건 어떨까요?

자동차 경주 미션에서 제어할 수 없는 랜덤값을 직접 사용하지 않고 외부에서 전략으로 주입받는 방식과 유사합니다
직접적인 호출(Random, now())을 피하고 외부 주입을 통해 테스트 하기 쉽게 만들어 볼 수 있어요! 😄

Reply:
고민했던 부분 중에 하나인데요.
테스트하기 유용한 부분이긴한데 요구사항에 없는데 하는게 맞을까 하는 생각을 했습니다.
테스트할 수 있게 되었는데 이제 이걸로 뭘 테스트를 해야할까하는 생각도 들었고요.
앞으로 날짜시간 같은게 필요하면 밖에서 전달 받는 구조로 짜야하는가하는 생각도 들었습니다.

어떤 접근 방법이 적절했다고 느끼셨나요?

Reply:
지금 구조에서는 딱히 테스트할 로직은 없습니다.
다만 다음과 같은 곳에서는 필요로 할 수 있어요

- 18:00 이후에는 예약을 받지 못한다.
이때 내부에서 시간을 생성한다면 테스트하기가 어려운 구조가 됩니다.

지금은 시간 관련한 요구사항이 없기 때문에 큰 장점은 못느끼는 것 같네요 😄 
참고만 하고 넘어가주세요~